torchchronos/download.py

```diff
@@ -1,40 +1,76 @@
+import os
 import shutil
 import tempfile
 import urllib.request
 import zipfile
 from pathlib import Path
-from typing import Optional
 
-from sktime.datasets._data_io import _download_and_extract, _list_available_datasets
+from sktime.datasets._data_io import _list_available_datasets
 from .typing import AnyPath
 
 
-def download_uea_ucr(extract_path: Optional[Path], dataset_name: str) -> None:
-    """This downloads the uea ucr dataset from sktime to the path extract_path."""
+from pathlib import Path
+import zipfile
+
+
+def _list_available_datasets(data_dir: AnyPath) -> list[str]:
+    datasets = []
+    for name in os.listdir(data_dir):
+        sub_dir = os.path.join(data_dir, name)
+        if os.path.isdir(sub_dir):
+            all_files = os.listdir(sub_dir)
+            if name + "_TRAIN.ts" in all_files and name + "_TEST.ts" in all_files:
+                datasets.append(name)
+    return datasets
+
+
+def download_uea_ucr(
+    extract_path: str | Path | None = None, dataset_name: str = "GunPoint"
+) -> bool:
+    """
+    Downloads the UEA/UCR time series dataset from sktime to the specified path.
+
+    Parameters:
+    extract_path (str | Path | None, default=None) : The path to extract the dataset. If None, extract to `.cache/data` directory.
+    dataset_name (str, default="GunPoint"): The name of the dataset to be downloaded.
```

Comment:
Usually, types & defaults are inferred from the signature (e.g. for renederingt docs) and do not need to be stated again.

Reply:
Usually is a bold statement,
this depends on which docstring standard you look at and even there it is not defnied explicitly, however chat gpt wrote it for me :D